intervals/interval.py

```diff
@@ -604,6 +604,16 @@ def __and__(self, other):
         lower = max(self.lower, other.lower)
         upper = min(self.upper, other.upper)
 
+        # If self and other were connected but lower > upper,
+        # then they must have been adjacent in which case there is
+        # no intersection and we return an empty interval
+        if lower > upper:
+            return self.__class__(
+                [upper, upper],
+                lower_inc=True,
+                upper_inc=False
+            )
```

Comment:
@bijanvakili had to do this, I can come back and clean up this code but for the purposes of unblocking us, is this ok?